Mailpoet 5796 inline notices

brezocordero opened the pull request:
## Description

This PR adds the sender domain notices to Automations.
These notices are only warnings as we don't block sending for Automations.

## Code review notes

_N/A_

## QA notes

To test:

Site needs to be connected to the shop with a valid key
Edit an automation that has a couple of email steps
Click on the email step
Check that you see the notices under the from email as the screenshot in the ticket
Authenticate a domain, check the notice disappears
Click on another email step
Check that if you add the same domain that you authenticated in the other step, you don't see the notice.

You can also check that the sender notices in Settings or inside of a newsletter/post notification have not changed.

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-5843]

## After-merge notes

_N/A_

## Tasks

- [ ] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [ ] I added sufficient test coverage
- [ ] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes


[MAILPOET-5843]: https://mailpoet.atlassian.net/browse/MAILPOET-5843?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

brezocordero committed:
Add Sender Domain Data to MailPoet Context

[MAILPOET-5796]

brezocordero committed:
Add reducer-action to update the context

Needed for automations with more than one email step where one step can authorize an email or sender domain. Not expected to be a common action.

[MAILPOET-5796]

brezocordero committed:
Add SenderDomainNotice to email Panel

[MAILPOET-5796]

brezocordero committed:
Adjust inline notice display for automations

[MAILPOET-5796]

brezocordero committed:
Fix PHP error warning on createAuthorizedSenderDomain

And reload cache

[MAILPOET-5796]

brezocordero committed:
Use props instead of window in inline notice

[MAILPOET-5796]

JanJakes commented on mailpoet/assets/js/src/automation/integrations/mailpoet/steps/send-email/edit/sender-domain-notice.tsx:
```diff
@@ -0,0 +1,87 @@
+    dispatch(storeName).alterContext('mailpoet', 'senderDomainsConfig', {
+      ...senderDomainsConfig,
+    });
+  }, [senderDomainsConfig]);
```
I noticed this component has a state that takes defaults from the context, and on change, it syncs them back to the context. Is it then necessary to have the state? Maybe a `useSelect()` on the context would work to read the value and the `dispatch` to update it? It seems like the state might be just a "copy".

brezocordero replied:
Thanks for the early feedback @JanJakes I really appreciate it.
I will follow your suggestion.

brezocordero replied:
Done in https://github.com/mailpoet/mailpoet/pull/5401/commits/52ca5a21194e500f80117e660bd61dcdff071303

brezocordero committed:
Use useSelect for senderDomainsConfig

[MAILPOET-5796]

brezocordero committed:
Add option to only show email rewritten warnings

This is for Automations where we don't block sending

[MAILPOET-5796]

brezocordero committed:
Modify context for Automations

[MAILPOET-5796]

websupporter commented on mailpoet/assets/js/src/automation/integrations/mailpoet/steps/send-email/edit/email-panel.tsx:
```diff
@@ -95,6 +97,12 @@ export function EmailPanel(): JSX.Element {
           )
         }
       />
+      {window.mailpoet_mss_active &&
```
Not a blocker, but why is this information in `window` and not in the context?

brezocordero replied:
It is always loaded in the window object in every page. Coming from [PageRenderer](https://github.com/mailpoet/mailpoet/blob/5cd7dad6268f0735082ee856b368dfe5e4afe615/mailpoet/lib/AdminPages/PageRenderer.php#L183).

websupporter committed:
Return early when senderDomainsConfig is null

The value can be null according to the definition. This happens when
another sending method is active or the API key is not able to send through
MSS.

[MAILPOET-5796]

websupporter commented on mailpoet/assets/js/src/automation/integrations/mailpoet/steps/send-email/edit/sender-domain-notice.tsx:
```diff
@@ -0,0 +1,87 @@
+function SenderDomainStatusNotice({
+  email,
+}: SenderDomainInlineNoticeProps): JSX.Element {
+  const [senderDomainsConfig, setSendersDomainConfig] = useState(
```
I get a blank screen with an API key from the creator plan. As far as I understand, that is because senderDomainsConfig can be `null` when the key is not able to send via MSS or another sending method is used. I pushed dc8ec6f47 to account for that configuration.

brezocordero replied:
Thanks for the fix

websupporter commented on mailpoet/lib/Services/AuthorizedSenderDomainController.php:
```diff
@@ -311,4 +311,25 @@ public function isAuthorizedDomainRequiredForNewCampaigns(): bool {
+      'allSenderDomains' => $this->getAllSenderDomains(),
+      'senderRestrictions' => [
+        'lowerLimit' => self::LOWER_LIMIT,
+        'upperLimit' => self::UPPER_LIMIT,
```
`upperLimit` seems not to be part of the TSX definition and not used anywhere. I wonder what this value is good for.

brezocordero replied:
It is a leftover from a first iteration. I have added it to the cleanup ticket.

websupporter reviewed (approved):
I found one error, which I fixed in dc8ec6f476238594fff3711087dfe90c3333061b. And I have two questions, but I think this can move forward to QA, as to me it seems this is nothing major and we want to release this change with the next release. @brezocordero can you check if my fix is correct?

brezocordero commented:
> can you check if my fix is correct?

It works thanks. I didn't know you could activate MSS with a creator key. Maybe a different issue.

Aschepikov commented:
Hi @brezocordero  I don't see the notice for the branded domain with no auth, in the settings I see it but in the automations no, can you take a look pls 
![image](https://github.com/mailpoet/mailpoet/assets/113185437/db654090-a616-4e5f-9494-10a8e427151b)

brezocordero commented:
> I don't see the notice for the branded domain with no auth, in the settings I see it but in the automations no, can you take a look pls

I can see it locally @Aschepikov . Is your key valid and for a plan with MSS?

Aschepikov commented:
Looks like a cache issue

Aschepikov reviewed (approved):
GJ!
